### PR TITLE
Add remove operation for table BBE

### PR DESCRIPTION
**Pull request by anupama-pathirage**
Fixes # https://github.com/ballerina-platform/ballerina-lang/issues/19592

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/table/table.bal [hunk 1]
```diff
@@ -8,37 +8,36 @@ type Employee record {
...
     // primary key in the column descriptor. Three data records are inserted into the `table`. The order of
     // the data values should match the order of the column descriptor.
     table<Employee> tbEmployee = table {
-        { key id, name, salary },
-        [ { 1, "Mary",  300.5 },
-          { 2, "John",  200.5 },
-          { 3, "Jim", 330.5 }
+        {key id, name, salary},
+        [
+            { 1, "Mary",  300.5},
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
            {1, "Mary",  300.5},
```

File: examples/table/table.bal [hunk 2]
```diff
@@ -8,37 +8,36 @@ type Employee record {
...
+            { 2, "John",  200.5},
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
            {2, "John",  200.5},
```

File: examples/table/table.bal [hunk 3]
```diff
@@ -8,37 +8,36 @@ type Employee record {
...
+            { 3, "Jim", 330.5}
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
            {3, "Jim", 330.5}
```

File: examples/table/table.bal [hunk 4]
```diff
@@ -74,10 +73,22 @@ public function main() {
...
     // This converts the `table` to JSON format.
     json retValJson = jsonutils:fromTable(tb);
-    io:println("JSON: ", retValJson);
+    io:println("JSON: ", retValJson.toJsonString());
 
     // This converts the `table` to XML format.
     xml retValXml = xmlutils:fromTable(tb);
     io:println("XML: ", retValXml);
+
+    //Removes employees who is having salary higher than 300.0
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
    // Removes employees with salaries higher than 300.0
```

File: examples/table/table.bal [hunk 5]
```diff
@@ -74,10 +73,22 @@ public function main() {
...
+    //from the table.
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
    // from the table.
```

File: examples/table/table.bal [hunk 6]
```diff
@@ -74,10 +73,22 @@ public function main() {
...
+    int | error count = tb.remove(isHigherSalary);
+    io:println("Deleted Count:", count);
+
+    //Contains employee records which is having salary less than 300.0
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
    // Contains records of employees with salaries less than 300.0.
```

File: examples/table/table.bal [hunk 7]
```diff
@@ -74,10 +73,22 @@ public function main() {
...
+    io:println(tb);
 }
 
+//Checks whether a given Employee's salary is higher than 300.0.
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
// Checks whether a given employee's salary is higher than 300.0.
```

File: examples/table/table.bal (hunk 1 above)
Comment (anupama-pathirage) [reply examples/table/table.bal]: Created issue https://github.com/ballerina-platform/ballerina-lang/issues/19595 to track this

File: examples/table/table.bal [hunk 8]
```diff
@@ -74,10 +73,22 @@ public function main() {
...
+    io:println("JSON: ", retValJson.toJsonString());
 
     // This converts the `table` to XML format.
     xml retValXml = xmlutils:fromTable(tb);
     io:println("XML: ", retValXml);
+
+    // Removes employees with salaries higher than 300.0
+    // from the table.
+    int | error count = tb.remove(isHigherSalary);
+    io:println("Deleted Count:", count);
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
    io:println("Deleted Count: ", count);
```

File: examples/table/table.bal [hunk 9]
```diff
@@ -74,10 +73,22 @@ public function main() {
...
+
+    // Contains records of employees with salaries less than 300.0.
+    io:println(tb);
 }
 
+// Checks whether a given employee's salary is higher than 300.0.
+function isHigherSalary(Employee p) returns (boolean) {
```
Comment (MaryamZi) [inline examples/table/table.bal]:
```suggestion
function isHigherSalary(Employee p) returns boolean {
```

File: examples/table/table.out [hunk 10]
```diff
@@ -15,5 +15,7 @@ Name: Jane
 Name: Anne
 Name: John
 Name: Peter
-JSON: id=1 name=Jane salary=300.5 id=2 name=Anne salary=100.5 id=3 name=John salary=400.5 id=4 name=Peter salary=150.0
+JSON: [{"id":1, "name":"Jane", "salary":300.5}, {"id":2, "name":"Anne", "salary":100.5}, {"id":3, "name":"John", "salary":400.5}, {"id":4, "name":"Peter", "salary":150.0}]
 XML: <results><result><id>1</id><name>Jane</name><salary>300.5</salary></result><result><id>2</id><name>Anne</name><salary>100.5</salary></result><result><id>3</id><name>John</name><salary>400.5</salary></result><result><id>4</id><name>Peter</name><salary>150.0</salary></result></results>
+2
```
Comment (MaryamZi) [inline examples/table/table.out]: this should have the "Deleted Count: " prefix right?

File: examples/table/table.bal (hunk 9 above)
Comment (MaryamZi) [reply examples/table/table.bal]: shall we also rename the variable? `emp` maybe?

**Pull request merged**